### PR TITLE
feat(api) new endpoints DeleteToken / DeleteTokenByUserId

### PR DESCRIPTION
**Pull request by a-launois**
## Description

New endpoints DeleteToken : 
```
DELETE /administration/tokens/{tokenName}
DELETE /administration/tokens/{tokenName}/users/{userId}
```

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by cg-tw — APPROVED** [review]

File: centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+DeleteTokenByUserId:
+  methods: DELETE
+  path: /administration/tokens/{tokenName}/users/{userId}
+  requirements:
+    tokenName: '.*'
```
Comment (abuathier) [inline centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml:5]: The token cannot be empty ? At least `.+` 
👉🏻 But I guess we can be more precise like for example `\w{20,30}`

File: centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    userId: '\d+'
+  controller: 'Core\Security\Token\Infrastructure\API\DeleteToken\DeleteTokenController'
+  condition: "request.attributes.get('version') >= 23.10"
+
+DeleteToken:
+  methods: DELETE
+  path: /administration/tokens/{tokenName}
+  requirements:
+    tokenName: '.*'
```
Comment (abuathier) [inline centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml:14]: Same as above

File: centreon/src/Core/Security/Token/Application/UseCase/DeleteToken/DeleteToken.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(TokenException::deleteNotAllowed())
+                );
+
+                return;
+            }
+
+            if ($userId === null) {
+                $userId = $this->user->getId();
+            }
```
Comment (abuathier) [inline centreon/src/Core/Security/Token/Application/UseCase/DeleteToken/DeleteToken.php:72]:
```suggestion
            $userId ??= $this->user->getId();
```

File: centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+DeleteTokenByUserId:
+  methods: DELETE
+  path: /administration/tokens/{tokenName}/users/{userId}
+  requirements:
+    tokenName: '.*'
+    userId: '\d+'
+  controller: 'Core\Security\Token\Infrastructure\API\DeleteToken\DeleteTokenController'
+  condition: "request.attributes.get('version') >= 23.10"
+
+DeleteToken:
```
Comment (abuathier) [inline centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml:10]:
```suggestion
# This is the same usecase as above but take the current user in place of {userId}
DeleteToken:
```

File: centreon/src/Core/Security/Token/Application/UseCase/DeleteToken/DeleteToken.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                $presenter->setResponseStatus(new NotFoundResponse('Token'));
+
+                return;
+            }
+
+            if (
+                ! $this->user->isAdmin()
+                && $token->getUserId() !== $this->user->getId()
+                && ! $this->user->hasRole(Contact::ROLE_MANAGE_TOKENS)
+            ) {
```
Comment (abuathier) [inline centreon/src/Core/Security/Token/Application/UseCase/DeleteToken/DeleteToken.php:88]: I have difficulties to read/understand this condition.
Maybe too much negations, if it is just me, don't mind and resolve.
```suggestion
            if (! (
                $this->user->isAdmin()
                || $token->getUserId() === $this->user->getId()
                || $this->user->hasRole(Contact::ROLE_MANAGE_TOKENS)
            )) {
```

File: centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml (hunk 1 above)
Comment (a-launois) [inline centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml:5]:
```suggestion
    tokenName: '.+'
```

File: centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml (hunk 2 above)
Comment (a-launois) [inline centreon/src/Core/Security/Token/Infrastructure/API/DeleteToken/DeleteTokenRoute.yaml:14]:
```suggestion
    tokenName: '.+'
```

**Review by cg-tw — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**